_pages/guides/data-hub-framework.markdown

```diff
@@ -5,17 +5,110 @@ lead_text: ''
+
+You can accomplish this recommended approach by moving a subset of files from inside the `marklogic` directory of your Grove project into your Data Hub project. Grove projects generated using the grove-cli's `grove new` command ship with this `marklogic` directory, which contains configuration files for a standalone MarkLogic database, optimized for Grove's sample data set.  Once the relevant files are migrated, the `marklogic` directory can be deleted entirely.  Additionally, an environment variable will need to be updated to reflect the data-hub-FINAL HTTP app server as configured by the Data Hub project.
+
+The advantages of this approach include: 
+- Combining the Grove-related artifacts with the Data Hub simplifies development and deployments.  The Grove code is logically separated inside its own directory under `data-hub/src/main/ui-modules` for clarity and distinction from Data Hub specific modules.
+- If code is maintained together in the same repository, consistent versioning can be ensured.
+- The Grove UI becomes a downstream consumer of the Data Hub.
+
+There are some downsides to this approach:
+- UI specific code will reside in close proximity to backend specific configurations.  This might not be as preferable to some development teams.
+- Any CRUD operations performed by the Grove project will write to the Data Hub's FINAL database.
+
+
+### How to implement the recommended approach, step-by-step:
+
+Prerequisite: MarkLogic Data Hub has been deployed to your MarkLogic host.  
+
+[Optional] If you previously installed Grove to its default database via the `mlDeploy` command within the `marklogic` directory, remove the Grove-specific configs before proceeding:
+        
+    > cd grove/marklogic
+    > ./gradlew mlUndeploy
+    
+
+### Steps to move Grove modules to the Data Hub
+
+1. Determine the MarkLogic host and port the data-hub-FINAL app server is running on.
+2. At the command line, from within the top level of your Grove project, (the directory with `middle-tier` and `ui` directories inside it), run `grove config` and enter the host and port of the data-hub-FINAL app server.  This will update environment variables for your project.
+3. Create the `ui-modules` directory inside the Data Hub project: `data-hub/src/main/ui-modules`.
+4. Update the Data Hub's `gradle.properties` to register this new location for MarkLogic module deployments.  Add the following line to `gradle.properites`:
+
+    ```
+    mlModulePaths=src/main/ml-modules,src/main/ui-modules
+    ```
+
+5. __Only necessary for Data Hub 5.0.1__ Update the Data Hub's `build.gradle` to use a newer version of ml-gradle.  Add the `dependencies` to the `buildscript` object around line 5: 
+
+    ```JSON
+    dependencies {
+		  classpath "gradle.plugin.com.marklogic:ml-gradle:3.14.0"
+    }
+    ```
```

Comment:
You could reword this slightly more generically, and tell everyone to check the ml-gradle version, saying it must be 3.14.0 or higher. (and a mention of 5.0.1 could be a side note "should only be necessary for dhf 5.0.1, or older dhf projects")

Reply:
Updated for proper version requirements.